crates/bindings-csharp/Runtime/Internal/FFI.cs

```diff
@@ -183,13 +209,10 @@ uint message_len
     );
 
     [LibraryImport(StdbNamespace)]
-    public static partial uint _buffer_len(Buffer buf_handle);
-
-    [LibraryImport(StdbNamespace)]
-    public static partial void _buffer_consume(
-        Buffer buf_handle,
-        [MarshalUsing(CountElementName = nameof(dst_len))] [Out] byte[] dst,
-        uint dst_len
+    public static unsafe partial short _bytes_source_read(
+        BytesSource source,
+        byte* buffer_ptr,
```

Comment:
I think you can use `Span<byte>` here and then you don't need to mark this function as `unsafe`, and the caller doesn't need to do the `unsafe`+`fixed` dance either - `LibraryImport` attribute takes care of that during marshalling.